src/qt/receivecoinsdialog.cpp

```diff
@@ -101,10 +101,18 @@ void ReceiveCoinsDialog::setModel(WalletModel *_model)
         }
 
         // eventually disable the main receive button if private key operations are disabled
-        ui->receiveButton->setEnabled(!model->privateKeysDisabled());
+        ui->receiveButton->setEnabled(model->canGetAddresses());
+
+        // Show new address button when keypool has new keys
```

Comment:
Show -> Enable? And maybe say "when new addresses can be generated by the wallet (i.e. when keypool has new keys)"